### PR TITLE
Create user image test

### PR DESCRIPTION
**Pull request by omus**
Adds a test for the code changes made in #21

### CODE REVIEW

**Comment by omus** [comment]
Note there is a script which builds system images called `build_sysimg.jl` but it will reside in "$JULIA_HOME/../share/julia/" or "$JULIA_HOME/../../contrib/" depending of if the version of Julia is official built or was built from source. I ended up not using this script as doing it by hand resulted in less pathing issues.

**Comment by omus** [comment]
Note I'm specifically testing for the issue that was fixed by #21. In that issue I only need to make the output-o file to generate the exception. I'm still using running `cc` as that step is required to make the final user image. At that point `julia --sysimage=$userimg_lib` would start Julia with the new system image.

**Comment by omus** [comment]
Sibling commit test which is expected to fail: https://travis-ci.org/invenia/Memento.jl/builds/227729800

**Comment by omus** [comment]
User image test is no longer in the allowed failures list. The sibling commit worked correctly when the user image test was expected to fail.

File: .travis/test.sh [hunk 1]
```diff
@@ -14,4 +14,13 @@ elif [[ "$TEST_TYPE" == "bench" ]]; then
     export MEMENTO_CURR_COMMIT="HEAD"
     export MEMENTO_BASE_COMMIT="origin/master"
     julia -e 'Pkg.test("Memento"; coverage=true)'
+elif [[ "$TEST_TYPE" == "userimage" ]]; then
```
Comment (rofinn) [inline .travis/test.sh]: Might be worth adding a few comments as this is a bit of a weird test case that most packages won't be doing.

**Pull request merged**